google-cloud-core-http/src/main/java/com/google/cloud/http/HttpTransportOptions.java

```diff
@@ -163,22 +163,18 @@ public void initialize(HttpRequest httpRequest) throws IOException {
         }
 
         HttpHeaders headers = httpRequest.getHeaders();
-        headers.set("x-goog-api-client", getXGoogApiClientHeader(serviceOptions));
+        headers.set(
+            "x-goog-api-client", getXGoogApiClientHeader(serviceOptions.getLibraryVersion()));
       }
     };
   }
 
-  String getXGoogApiClientHeader(ServiceOptions<?, ?> serviceOptions) {
-    return String.format(
+  String getXGoogApiClientHeader(String libraryVersion) {
+    return String.format(Locale.US,
         "gl-java/%s %s/%s",
-        getJavaVersion(),
+        firstNonNull(Runtime.class.getPackage().getImplementationVersion(), ""),
```

Comment:
I removed the method, because otherwise I would have to have the same method in two places (GrpcTransportOptions and HttpTransportOptions). Having two methods which return "java version" (somthing constant for same instance of an application) seems weird. We could also make that a public method somewhere and reuse it, but it seems unnecessary (trying to not add anything to public api unless it is necessary).